### PR TITLE
Refactored hook handlers in mod_bosh

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `mod_bosh` module to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/148697) / small_tests / 747caa624d1c4b0508c5d6f1553bc294d61081c3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3840/148697/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3840/148697/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-03_12.58.18/lib.mongooseim.logs/run.2022-11-03_12.58.19/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/148747) / small_tests / 5cdc94ebdc420326d4d1ddcaa05023267e7044e4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3840/148747/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3840/148747/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-03_13.51.03/lib.mongooseim.logs/run.2022-11-03_13.51.04/suite.log.html)

File: src/mod_bosh.erl [hunk 1]
```diff
@@ -89,7 +89,7 @@ start(_HostType, Opts) ->
         false ->
             mod_bosh_backend:start(Opts),
             {ok, _Pid} = mod_bosh_socket:start_supervisor(),
-            ejabberd_hooks:add(node_cleanup, global, ?MODULE, node_cleanup, 50)
+            gen_hook:add_handler(node_cleanup, global, fun ?MODULE:node_cleanup/3, #{}, 50)
```
Comment (kamilwaz) [inline src/mod_bosh.erl:92]: This hook should be unregistered when the module is unloaded. I recommend to move it to `hooks/1` and add the following code at start/stop functions:
```
gen_hook:add_handlers(hooks(HostType)) % start
gen_hook:delete_handlers(hooks(HostType)) % stop
```

The specification for `hooks/1`: `-spec hooks(mongooseim:host_type()) -> gen_hook:hook_list().`.

Comment (pawlooss1) [reply src/mod_bosh.erl]: done 👍

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/149135) / small_tests / 2f41f24f9be258fdf2f48aedc330a427ad46d18e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3840/149135/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3840/149135/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-07_08.50.56/lib.mongooseim.logs/run.2022-11-07_08.50.57/suite.log.html)

**Review by kamilwaz — APPROVED** [review]

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/149163) / small_tests / 7ea2ea9d607fc7c7bd13c87449d3bc18e97050f0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3840/149163/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3840/149163/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-07_09.23.24/lib.mongooseim.logs/run.2022-11-07_09.23.25/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/149203) / small_tests / 360ce937772c3662f266056142dec93ad4e27ddc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3840/149203/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3840/149203/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-07_09.54.25/lib.mongooseim.logs/run.2022-11-07_09.54.25/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/149208) / small_tests / bef6a1926936655111b3d442247962ca91c871f5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3840/149208/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3840/149208/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-07_09.55.09/lib.mongooseim.logs/run.2022-11-07_09.55.09/suite.log.html)

**Review by NelsonVides — APPROVED** [review]
LGTM

**Pull request merged**